superset/connectors/sqla/models.py

```diff
@@ -632,8 +643,28 @@ def get_sqla_query(  # sqla
             time_filters.append(dttm_col.get_time_filter(from_dttm, to_dttm))
 
         select_exprs += metrics_exprs
+
+        labels_expected = [str(c.name) for c in select_exprs]
+
+        # Some databases will just return the group-by field into the select, but don't
+        # allow the group-by field to be put into the select list.
+        if not columns and db_engine_spec.groupby_fields_auto_present_in_select:
+            select_sans_groupby = []
+            # We want identity and not equality, so doing the filtering manually
+            for s in select_exprs:
+                for g in groupby_exprs:
+                    if s is g:
+                        break
+                else:
+                    select_sans_groupby.append(s)
+            select_exprs = select_sans_groupby
+
```

Comment:
Perhaps this could be moved to `db_engine_specs.py` to a method along the lines of `def make_select_compatible(select_exprs, group_exprs)`, which for `BaseEngineSpec` returns `select_exprs`and in the case of Pinot performs the above logic. Check `mutate_label(label)` in `db_engine_specs.py` for an example.